src/test/ui/async-await/zero-uses-in-fn.rs

```diff
@@ -0,0 +1,19 @@
+// build-pass
+// edition:2018
+
+// Avoid spurios warnings of unused lifetime. The below async functions
+// are desugered to have an unused lifetime
+// but we don't want to warn about that as there's nothing they can do about it.
+
+#![deny(unused_lifetimes)]
+#![allow(dead_code)]
+
+pub async fn october(s: &str) {
+    println!("{}", s);
+}
+
+pub async fn async_fn(&mut ref s: &mut[i32]) {
+    println!("{:?}", s);
+}
```

Comment:
Could we add a test case for a "correct" lint here? i.e., warning on an actually unused lifetime in async fn?

Reply:
Excellent idea.